game/scoring.cpp

```diff
@@ -0,0 +1,129 @@
+#include "yahtzee.h"
+#include <algorithm>
+// Util functions
+int sum(int die[5]) {
+    int sum = 0;
+    for (int i = 0; i < 5; ++i)
+    {
+        sum += die[i];
+    }
+    return sum;
+}
+
+//Funtions for scoring yahtzee hand
+
+// Out is an array initialized to zero in which to hold the count
+int sortDie(int die[5], int out[6]) {
+    for (int i = 0; i < 5; ++i)
+    {
+        out[die[i]-1] += 1;
+    }
+    return 0;
```

Comment:
At the end of all this, the signature of this function should be `void sortDie(std::vector<int>* die)`